os/arch/arm/src/armv8-m/up_unblocktask_withoutsavereg.c

```diff
@@ -136,17 +136,17 @@ void up_unblock_task_without_savereg(struct tcb_s *tcb)
 		tcb->task_state = TSTATE_TASK_RUNNING;
 		tcb->flink->task_state = TSTATE_TASK_READYTORUN;
 
-#ifdef CONFIG_ARMV8M_TRUSTZONE
-		if (tcb->tz_context) {
-			TZ_StoreContext_S(tcb->tz_context);
-		}
-#endif
 	} else {
 		/* The new btcb was added in the middle of the ready-to-run list */
 
 		tcb->task_state = TSTATE_TASK_READYTORUN;
 	}
 
+#ifdef CONFIG_ARMV8M_TRUSTZONE
+	if (rtcb->tz_context) {
+		TZ_StoreContext_S(rtcb->tz_context);
+	}
+#endif
```

Comment:
I think it should be called if a head of list is changed to tcb (in line 131-138), isn't it?

Reply:
Even if the head is not changed, we running the below context switch code.
In that case we will call load_context() always. 
So, corresponding call to store_context() also needs to be done always.